internal/extender/resource.go

```diff
@@ -304,14 +306,27 @@ func (s *SparkSchedulerExtender) selectDriverNode(ctx context.Context, driver *v
 			return "", failureEarlierDriver, werror.Error("earlier drivers do not fit to the cluster")
 		}
 	}
-	driverNode, executorNodes, hasCapacity := s.binpacker.BinpackFunc(
+
+	// run binpack with the configured function, usually AZ tight pack
+	packingResultDefault := s.binpacker.BinpackFunc(
+		ctx,
+		applicationResources.driverResources,
+		applicationResources.executorResources,
+		applicationResources.minExecutorCount,
+		driverNodeNames,
+		executorNodeNames,
+		availableNodesSchedulingMetadata)
+	driverNode, executorNodes, hasCapacity := packingResultDefault.DriverNode, packingResultDefault.ExecutorNodes, packingResultDefault.HasCapacity
+	// as an experiment, binpack with tight pack (report in metrics but don't use this result)
+	packingResultTight := SelectBinpacker("tightlyPack").BinpackFunc(
```

Comment:
I think we can do this but lets keep the PR tightly scoped and do this in another PR

Reply:
Ack!